src/kernels/kernelFinder.base.ts

```diff
@@ -128,19 +130,24 @@ export abstract class BaseKernelFinder implements IKernelFinder {
     ): Promise<KernelConnectionMetadata[]> {
         this.startTimeForFetching = this.startTimeForFetching ?? new StopWatch();
 
-        try {
-            // Get both local and remote kernels.
-            const [localKernels, remoteKernels] = await Promise.all([
-                this.listLocalKernels(resource, cancelToken, useCache),
-                this.listRemoteKernels(resource, cancelToken, useCache)
-            ]);
+        // Get both local and remote kernels.
+        const [localKernels, remoteKernels] = await Promise.all([
+            this.listLocalKernels(resource, cancelToken, useCache).catch((ex) => {
+                traceError('Failed to get local kernels', ex);
+                return [];
+            }),
+            this.listRemoteKernels(resource, cancelToken, useCache).catch((ex) => {
+                traceError('Failed to get remote kernels', ex);
+                // When remote kernels fail, turn off remote if we get a ECONNREFUSED error
```

Comment:
Not sure this is the right thing.
Shouldn't this be an action performed by the user, what if there's no net work connection.
I think removing somethingn the user added is best performed by the user. I.e. we can still try and fail, but removing should be left to the user.

Reply:
This feels like a change in behaviour today, if you're fixing a regressino that was introduced in the refactor thats fine, 
however I doubt we removed these items from the list in the past, & that feels like a change thats not necessary to fix this bug.

Reply:
Without this change if the remote fails, you can never start a kernel. Because it keeps picking the remote URI as the 'active' interpreter for the interactive window.

See this code here:
https://github.com/microsoft/vscode-jupyter/blob/a912533e456111d267f6d3838936e610387bbc03/src/notebooks/controllers/notebookControllerManager.ts#L183

URI isn't removed, it's just not used anymore. So if you bring up the picker again, it's in the MRU list